providers/provider_data.go

```diff
@@ -26,21 +26,17 @@ const (
 // ProviderData contains information required to configure all implementations
 // of OAuth2 providers
 type ProviderData struct {
-	ProviderName      string
-	LoginURL          *url.URL
-	RedeemURL         *url.URL
-	ProfileURL        *url.URL
-	ProtectedResource *url.URL
-	ValidateURL       *url.URL
-	// Auth request params & related, see
-	//https://openid.net/specs/openid-connect-basic-1_0.html#rfc.section.2.1.1.1
-	AcrValues        string
-	ApprovalPrompt   string // NOTE: Renamed to "prompt" in OAuth2
-	ClientID         string
-	ClientSecret     string
-	ClientSecretFile string
-	Scope            string
-	Prompt           string
+	ProviderName       string
+	LoginURL           *url.URL
+	LoginURLParameters map[string]options.LoginURLParameter
```

Comment:
So this is a weird pattern within this codebase, normally we try to create relatively small, self contained packages with new code that don't all depend on the larger options package. I would have expected a local struct which contains the default and the compiled regex, constructed when we construct the provider data.